src/test/java/yapchit/HandlerTest.java

```diff
@@ -0,0 +1,79 @@
+package yapchit;
+
+import yapchit.tasks.Deadline;
+import yapchit.yapchitexceptions.InvalidDetailException;
+import yapchit.yapchitexceptions.YapchitException;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+public class HandlerTest {
+
+    private Handler handler;
+    private TaskList tasks;
+    private Ui ui;
+    private Parser parser;
+
+    public HandlerTest() {
+        handler = new Handler();
+        tasks = new TaskList();
+        ui = new Ui();
+        parser = new Parser();
+    }
+
+    @Test
+    void testHandleDeadlineSuccess() throws YapchitException {
```

Comment:
Suggestion to name test cases according to the java coding standard:
```suggestion
    void handleDeadline_variedSpaces_success() throws YapchitException {
```